pkg/agent/datamodel/helper.go

```diff
@@ -48,6 +48,32 @@ func IsMIGNode(gpuInstanceProfile string) bool {
 	return gpuInstanceProfile != ""
 }
 
+// IsNvidiaEnabledSKU determines if an VM SKU has nvidia driver support.
+func IsNvidiaEnabledSKU(vmSize string) bool {
+	// Trim the optional _Promo suffix.
+	vmSize = strings.ToLower(vmSize)
+	vmSize = strings.TrimSuffix(vmSize, "_promo")
+	return NvidiaEnabledSKUs[vmSize]
+}
```

Comment:
Can you also bring over the test coverage from karpenter for all of these functions? Its really barebones but still useful a bit https://github.com/Azure/karpenter-provider-azure/blob/main/pkg/utils/gpu_test.go

Reply:
Moved over. Also merged with some tests from RP.